fastestimator/network.py

```diff
@@ -176,45 +154,82 @@ def __init__(self, ops: Iterable[Union[TensorOp, Scheduler[TensorOp]]]):
         self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
 
     def load_epoch(self, mode: str, epoch: int) -> List[TensorOp]:
-        epoch_ops = super().load_epoch(mode, epoch)
+        super().load_epoch(mode, epoch)
         if self.device.type == "cuda":
             for model in self.epoch_models:
                 model.to(self.device)
-        return epoch_ops
+        update_ops = [op for op in self.epoch_ops if isinstance(op, UpdateOp)]
+        for idx, update_op in enumerate(update_ops):
```

Comment:
we can drop this and use .detach() when collecting the tensors at the end instead